pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -60,6 +60,12 @@ func (r *pullthroughBlobStore) remoteStat(ctx context.Context, dgst digest.Diges
 		return distribution.Descriptor{}, err
 	}
 
+	r.pullFromInsecureRegistries = false
```

Comment:
> An incorrect value can be set upon creation.

I don't get it. Decision made on upper level shall be respected unless overridden by image stream annotation. If there's an incorrect value passed, it should be fixed in upper levels.